src/diffusers/models/attention.py

```diff
@@ -377,6 +379,8 @@ def __init__(
             self.norm3 = nn.LayerNorm(dim, norm_eps, norm_elementwise_affine)
         elif norm_type == "layer_norm_i2vgen":
             self.norm3 = None
+        else:
+            self.norm3 = nn.LayerNorm(dim, norm_eps, norm_elementwise_affine)
```

Comment:
Yes, Latte uses `ada_norm_single`. However, there are some differences between the Latte model and Pixartalpha that make it impossible to call `norm2` directly in the forward function to replace `norm3`. Pixartalpha is a text-to-image model, so there is only one Transformer block in its model structure, which contains two multi-head attention. `norm2` acts on a second multi-head attention mechanism. But there are two Transformer blocks in the Latte, the former containing two multi-head attention mechanisms and the latter containing only one multi-head attention. Therefore, in the second Transformer block, there is no definition of `norm2`. So we can't recall `norm2` in the forward function to replace `norm3`.


BTW, I noticed that `norm_elementwise_affine` for `nn.LayerNorm` is set as `False`. Therefore, for Latte, all three norms in the Transformer block are the same (`nn.LayerNorm` is set to unlearnable).

In forward function, Latte goes to line 516.
![image](https://github.com/huggingface/diffusers/assets/38418898/aa7dd806-9369-4ef7-8e39-2e511f52423f)

Reply:
I think this is what happens for pixart
                                  
                           norm1 -> attn1 -> attn2 -> norm2

what is latte's? if you can provide an example testing script that we can run latte with the current implementation under this PR, it will be helpful too

Reply:
ok I got it! for the second bloc `cross_attention_dim=None`

Reply:
ok I think we can do here in this bloc, https://github.com/huggingface/diffusers/blob/25d7bb3ea65bb6f79056252584e24eb545c5eb18/src/diffusers/models/attention.py#L333
```python 
if cross_attention_dim is not None or double_self_attention:
    ...
else:
   if norm_type == "ada_norm_single":
      self.norm2 = nn.LayerNorm(...)
   else: 
    self.norm2=None
  self.attn2 = None
```

I don't think pixart use self-attention blocks so it should not affect it here, but we would want to double check  cc @sayakpaul too

Reply:
Yes, the `cross_attention_dim` in the second Transfomer block is `None`. The second Transformer block for `Latte` only contains the `attn1` and dose not have `attn2`.

Reply:
so this should work, no? https://github.com/huggingface/diffusers/pull/8404#discussion_r1646876718

Reply:
Yes, I think this manner should work and does not affect others.

Specifically,
![image](https://github.com/huggingface/diffusers/assets/38418898/92535ed0-823e-416b-81ff-62988ed3a877)

and 

![image](https://github.com/huggingface/diffusers/assets/38418898/122cf118-fcfa-43ad-81b6-3497e324183a)

Reply:
I meant the solution I proposed in this comment https://github.com/huggingface/diffusers/pull/8404#discussion_r1646876718
I think this should work and this is the only change needed for here

Reply:
so basically you will have norm1 -> attn1 -> norm2 for latte

Reply:
Yes, We will have two norms in the temporal Transformer for Latte. However, I looked at the naming convention for `diffusers`. `norm1` acts on `attn1`, `norm2` acts on `attn2`, and `norm3` acts on `MLP`. Therefore, I think it is more consistent with the `diffusers` naming convention to skip `norm2` and `attn2`. Because for Latte's temporal Transformer, its structure is `norm`->`attn`->`norm`->`mlp`.


And from this picture, I don't quite understand what you mean by the method. Do you mean we can change `norm2 `here for Latte?
![image](https://github.com/huggingface/diffusers/assets/38418898/edcfc355-01bb-476d-b3a4-495eae466497)

Reply:
yes basically here https://github.com/huggingface/diffusers/blob/a9c403c00197d8d6eb854128dda2f7849cedd100/src/diffusers/models/attention.py#L361

currently is 
```python
else:
    self.norm1 = None
    self.attn2=None
```


you can change it to 

```
else:
   if norm_type == "ada_norm_single":
      self.norm2 = nn.LayerNorm(...)
   else: 
    self.norm2=None
  self.attn2=None
```

Reply:
Okay. I have modified `norm2` according to your suggestion.

Reply:
> I think this is what happens for pixart
> 
> ```
>                        norm1 -> attn1 -> attn2 -> norm2
> ```
> 
> what is latte's? if you can provide an example testing script that we can run latte with the current implementation under this PR, it will be helpful too

Sorry. I missed this message. You can run Latte using the following script:

```
from diffusers import LattePipeline
from diffusers.models import AutoencoderKLTemporalDecoder
from torchvision.utils import save_image
import torch
import imageio

torch.manual_seed(0)

device = "cuda" if torch.cuda.is_available() else "cpu"
video_length = 16
pipe = LattePipeline.from_pretrained("/mnt/hwfile/gcc/maxin/work/pretrained/Latte/", torch_dtype=torch.float16).to(device)
# vae = AutoencoderKLTemporalDecoder.from_pretrained("/mnt/hwfile/gcc/maxin/work/pretrained/Latte/", subfolder="vae_temporal_decoder", torch_dtype=torch.float16).to(device)
# pipe.vae = vae

prompt = "a cat wearing sunglasses and working as a lifeguard at pool."
videos = pipe(prompt, video_length=video_length).video.cpu()


if video_length > 1:
    videos = (videos.clamp(0, 1) * 255).to(dtype=torch.uint8) # convert to uint8
    imageio.mimwrite('./test_pipeline.mp4', videos[0].permute(0, 2, 3, 1), fps=8, quality=5) # highest quality is 10, lowest is 0
else:
    save_image(videos[0], './test_pipeline.png')
```